svm/tests/integration_test.rs

```diff
@@ -141,32 +142,72 @@ impl SvmTestEnvironment<'_> {
         // with SIMD83, an account might change multiple times in the same batch
         // but it might not exist on all transactions
         let mut final_accounts_actual = self.test_entry.initial_accounts.clone();
+        let update_or_dealloc_account =
+            |final_accounts: &mut AccountsMap, pubkey, account: AccountSharedData| {
+                if account.lamports() == 0 {
+                    final_accounts.insert(pubkey, AccountSharedData::default());
+                } else {
+                    final_accounts.insert(pubkey, account);
+                }
+            };
+
+        for (tx_index, processed_transaction) in batch_output.processing_results.iter().enumerate()
+        {
+            let sanitized_transaction = &transactions[tx_index];
 
-        for (index, processed_transaction) in batch_output.processing_results.iter().enumerate() {
             match processed_transaction {
                 Ok(ProcessedTransaction::Executed(executed_transaction)) => {
                     for (pubkey, account_data) in
                         executed_transaction.loaded_transaction.accounts.clone()
                     {
-                        final_accounts_actual.insert(pubkey, account_data);
+                        let is_writable = sanitized_transaction
+                            .account_keys()
+                            .iter()
+                            .position(|key| key == &pubkey)
```

Comment:
can we not `enumerate` accounts above to get index? this is `O(n^2)`
(not a huge deal since this is test only)

Reply:
oh yes youre right, i wasnt thinking about the fact that `LoadedTransactionAccounts` has the same ordering as `SanitizedTransaction`